data/tabular/train_test_split.py

```diff
@@ -1,209 +1,459 @@
+        # and write the new file to `data_clean.csv`
+        if not override:
+            os.rename(
+                os.path.join(os.path.dirname(file), "data_clean.csv"),
+                os.path.join(os.path.dirname(file), "data_clean_old.csv"),
+            )
+        df.to_csv(
+            os.path.join(os.path.dirname(file), "data_clean.csv"),
+            index=False,
+        )
+
+
+def smiles_split(
+    data_dir,
     override: bool = False,
-    check: bool = True,
+    seed: int = 42,
+    debug: bool = False,
+    train_frac: float = 0.7,
+    val_frac: float = 0.15,
+    test_frac: float = 0.15,
+    val_smiles_path: Union[str, Path] = "val_smiles.txt",
+    test_smiles_path: Union[str, Path] = "test_smiles.txt",
 ):
-    paths_to_data = glob(path)
-
-    # uncomment the following lines for debugging on a subset of data
-    # filtered_paths = []
-    # for path in paths_to_data:
-    #     if "flashpoint" in path:
-    #         filtered_paths.append(path)
-    #     elif "BBBP" in path:
-    #         filtered_paths.append(path)
-    #     elif "peptide" in path:
-    #         filtered_paths.append(path)
-    #     elif "bicerano_dataset" in path:
-    #         filtered_paths.append(path)
-    # paths_to_data = filtered_paths
-
-    representations = []
-
-    for path in paths_to_data:
-        df = pd.read_csv(path)
-
-        if repr_col in df.columns and path not in TRACKED_DATASETS:
-            print("Processing", path.split("/")[0])
-            TRACKED_DATASETS.append(path)
-            representations.extend(df[repr_col].to_list())
-
-    repr_df = pd.DataFrame()
-    repr_df[repr_col] = list(set(representations))
-
-    if "SMILES" in repr_df.columns:
-        split = create_scaffold_split(
-            repr_df, seed=seed, frac=[train_size, val_size, test_size]
+    # make deterministic
+    np.random.seed(seed)
+    random.seed(seed)
+    dask_random_state = da.random.RandomState(seed)
+
+    def assign_split(
+        row,
+        test_smiles,
+        val_smiles,
+        train_frac,
+        test_frac,
+        random_state=dask_random_state,
+    ):
+        if row["SMILES"] in test_smiles:
+            return "test"
+        elif row["SMILES"] in val_smiles:
+            return "valid"
+        else:
+            # Assign random number for train/remaining split using the random state
+            random_value = random_state.random_sample()
+            if random_value < train_frac:
+                return "train"
+            elif random_value < train_frac + test_frac:
+                return "test"
+            else:
+                return "valid"
+
+    # we err toward doing more I/O but having simpler code to ensure we don't make anything stupid
+    all_yaml_files = get_yaml_files(data_dir)
+    smiles_yaml_files = [
+        file for file in all_yaml_files if has_yaml_file_smiles_column(file)
+    ]
+    # we filter those out that are in the to_scaffold_split list
+    not_scaffold_split_yaml_files = [
```

Comment:
why `not` in var name, this contains the entries for the scaffold split, or am I wrong?

Reply:
ah, I get it, this is then to apply it those not used for the scaffold split